SportsHub/SportsHub.Api/Exceptions/ExceptionHandler.cs

```diff
@@ -10,11 +11,34 @@ public async Task InvokeAsync(HttpContext context, RequestDelegate next)
         {
             await next(context);
         }
-
-        catch (Exception e)
+        catch (BusinessLogicException ex)
+        {
+            await HandleBusinessLogicExceptionAsync(context, ex);
+        }
+        catch (NotFoundException ex)
+        {
+            await HandleNotFoundExceptionAsync(context, ex);
+        }
+        catch (Exception ex)
         {
             context.Response.StatusCode = (int)HttpStatusCode.InternalServerError;
-            await context.Response.WriteAsync(e.Message);
+            await context.Response.WriteAsync(ex.Message);
         }
     }
+
+    private async Task HandleNotFoundExceptionAsync(HttpContext context, NotFoundException ex)
+    {
+        context.Response.ContentType = "/application/json";
+        context.Response.StatusCode = ex.ErrorCode;
+        string result = ex.ToString();
+        await context.Response.WriteAsync(result);
+    }
+
+    private async Task HandleBusinessLogicExceptionAsync(HttpContext context, BusinessLogicException ex)
+    {
```

Comment:
what is different between these methods? HandleNotFoundExceptionAsync and current?

Reply:
I probably forgot to change it . Because before i tried to throw exceptions in business layer i didnt had the ExceptionMessages class and i was setting different messages in this methods for testing . After i got home from work i will see to it